src/main/java/com/tmtravlr/additions/gui/view/edit/GuiEditPotionType.java

```diff
@@ -0,0 +1,195 @@
+        this.useUniqueColorInput = new GuiComponentBooleanInput(I18n.format("gui.edit.potionType.potionColor.switch.label"), this);
+        // Reason for noinspection: readability. The suggested change was to `!this.isNew && this.addition.hasColor`
+        //noinspection SimplifiableConditionalExpression
+        this.useUniqueColorInput.setDefaultBoolean(this.isNew ? false : this.addition.hasColor());
+        this.potionColorInput = new GuiComponentColorInput(I18n.format("gui.edit.potionType.potionColor.label"), this);
+
+        // Add all components to the screen
+        this.components.add(this.idInput);
+        this.components.add(this.baseNameInput);
+        this.components.add(this.potionEffectListInput);
+
+        this.advancedComponents.add(this.inputInformation);
+        this.advancedComponents.add(this.splashNameInput);
+        this.advancedComponents.add(this.lingeringNameInput);
+        this.advancedComponents.add(this.arrowNameInput);
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Override
+    public void saveObject() {
+        String name = this.isNew ? this.addon.id + "-" + this.idInput.getText() : this.addition.id;
+        if (CommonGuiUtils.requireStringField(
+                this.idInput,
+                this,
+                "gui.edit.potionType.problem.title",
+                new TextComponentTranslation("gui.edit.problem.noId")
+        ) == null) return;
+        String baseName = CommonGuiUtils.requireStringField(this.baseNameInput, this, "gui.edit.potionType.problem.title", new TextComponentTranslation("gui.edit.potionType.problem.noBase"));
+        if (baseName == null) return;
+
+        String base = TextFormatting.getTextWithoutFormattingCodes(baseName), formats=baseName.substring(0, baseName.indexOf(base));
+        String splashName = this.splashNameInput.getText().isEmpty() ? formats + "Splash " + base : this.splashNameInput.getText();
+        String lingeringName = this.lingeringNameInput.getText().isEmpty() ? formats + "Lingering " + base : this.lingeringNameInput.getText();
+        String arrowName;
+        if (this.arrowNameInput.getText().isEmpty()) {
+            arrowName = baseName.startsWith("Potion of") ? baseName.replace("Potion of", "Arrow of") : formats + "Arrow of " + base;
+        } else {
+            arrowName = this.arrowNameInput.getText();
+        }
+        // No potion color for a unique potion type is possible without special modifications, but to
+        // avoid lots of special modifications, this still exists.
+        int potionColor = this.useUniqueColorInput.getBoolean() ? this.potionColorInput.getColorInt() : -1;
+
+        PotionEffect[] effects = new PotionEffect[this.potionEffectListInput.getComponents().size()];
+        List<GuiComponentPotionEffectInput> effectInputs = this.potionEffectListInput.getComponents();
+        for (int i = 0; i < effectInputs.size(); i++) {
+            effects[i] = effectInputs.get(i).getPotionEffect();
+        }
```

Comment:
You can use a stream for this!

Edit: Added a step to remove nulls
```suggestion
        PotionEffect[] effects = effectInputs.getComponents().stream()
                .map(GuiComponentPotionEffectInput::getPotionEffect)
                .filter(effect -> effect != null)
                .collect(Collectors.toList())
                .toArray(new PotionEffect[0]);
```